sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/date/BeamSqlTimestampMinusIntervalExpression.java

```diff
@@ -52,8 +50,8 @@ public boolean accept() {
   static boolean accept(List<BeamSqlExpression> operands, SqlTypeName outputType) {
     return operands.size() == 2
         && SqlTypeName.TIMESTAMP.equals(outputType)
-        && SqlTypeName.TIMESTAMP.equals(operands.get(0).getOutputType())
-        && INTERVALS_DURATIONS_TYPES.containsKey(operands.get(1).getOutputType());
+        && SqlTypeName.DATETIME_TYPES.contains(operands.get(0).getOutputType())
```

Comment:
This one line appears to be the only functional change in this diff. Took me a few minutes to find it because of all the no-op changes. So with this change you are expanding the set of supported types to `DATE, TIME, TIME_WITH_LOCAL_TIME_ZONE, TIMESTAMP, TIMESTAMP_WITH_LOCAL_TIME_ZONE`. I think you need to rename this class to `BeamSqlDatetimeMinusIntervalExpression` to reflect its new behavior.

Reply:
You're right. I need to separate it into `BeamSqlDatetimeMinusIntervalExpression` to handle all other datetime related types.